### PR TITLE
Add ResultRecord#doesNotHaveErrors and ResultRecord#getErrorMessageFor

### PR DESCRIPTION
**Pull request by mrjones2014**
- [ ] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/view-models/result-record.test.ts [hunk 1]
```diff
@@ -203,6 +207,82 @@ describe("ResultRecord", () => {
...
+
+    // -----------------------------------------------------------------------------------------
+    // #region getErrorMessageFor
+    // -----------------------------------------------------------------------------------------
+
+    describe("#getErrorMessageFor", () => {
+        it("When no error exists for given key, then returns undefined", () => {
+            // Arrange
+            const errorKey = "TEST_ERROR_KEY";
+            const resultRecord = ResultRecordFactory.build({
```
Comment (brandongregoryscott) [inline src/view-models/result-record.test.ts:258]: do these need to be statically referenced? I think we usually usually do `Factory.build<ResultRecord>(...)`

**Review by brandongregoryscott — COMMENTED** [review]
Just one question before approval @mrjones2014

File: src/view-models/result-record.test.ts (hunk 1 above)
Comment (mrjones2014) [reply src/view-models/result-record.test.ts:258]: I can make this change to be consistent with everywhere else.

Comment (mrjones2014) [reply src/view-models/result-record.test.ts:258]: I can make this change to be consistent with everywhere else.

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**